dozer-api/src/grpc/internal/internal_pipeline_server.rs

```diff
@@ -6,45 +6,83 @@ use crate::grpc::internal_grpc::{
 use crossbeam::channel::Receiver;
 use dozer_types::{crossbeam, log::info, models::app_config::Config, tracing::warn};
 use std::{net::ToSocketAddrs, pin::Pin};
-use tokio::runtime::Runtime;
-use tokio_stream::{wrappers::ReceiverStream, StreamExt};
+use tokio::{
+    runtime::Runtime,
+    sync::broadcast::{self, Sender},
+};
+use tokio_stream::wrappers::ReceiverStream;
 use tonic::{codegen::futures_core::Stream, transport::Server, Response, Status};
 
 pub struct InternalPipelineServer {
     app_config: Config,
-    receiver: Receiver<PipelineResponse>,
+    receiver: broadcast::Receiver<PipelineResponse>,
+}
+impl InternalPipelineServer {
+    pub fn new(app_config: Config, receiver: Receiver<PipelineResponse>) -> Self {
+        let (tx, rx1) = broadcast::channel::<PipelineResponse>(16);
+        tokio::spawn(async move {
+            Self::setup_broad_cast_channel(tx, receiver);
+        });
+        Self {
+            app_config,
+            receiver: rx1,
+        }
+    }
+
+    fn setup_broad_cast_channel(
+        tx: Sender<PipelineResponse>,
+        receiver: Receiver<PipelineResponse>,
+    ) {
+        loop {
+            let message = receiver.recv();
+            match message {
+                Ok(message) => {
+                    let result = tx.send(message);
+                    if let Err(e) = result {
+                        warn!("Error sending message to broadcast channel: {:?}", e);
+                    }
+                }
+                Err(err) => {
+                    warn!("Message reveived error: {:?}", err);
+                    break;
+                }
+            }
+        }
+    }
 }
 type ResponseStream = Pin<Box<dyn Stream<Item = Result<PipelineResponse, Status>> + Send>>;
 
 #[tonic::async_trait]
 impl InternalPipelineService for InternalPipelineServer {
     type StreamPipelineRequestStream = ResponseStream;
-
     async fn stream_pipeline_request(
         &self,
         _request: tonic::Request<PipelineRequest>,
     ) -> Result<Response<ResponseStream>, Status> {
         let (tx, rx) = tokio::sync::mpsc::channel(1000);
```

Comment:
This is the problem I believe. We should have channels that we use across pipeline to websocket comparable in terms of capacity. We probably need to give internal more capacity than what pipeline is configured. Ideally this should be adjustable through configuration @chubei @duonganhthu43 what do you think ? 

When i tested locally with higher capacity, eth messages were not dropped.

Reply:
Interesting. I thought we wound block on channel becoming full. I think we should find why channel capacity affects the program behaviour becuaes otherwise backpressure won't be working properly.

Reply:
we could try `unbounded_channel` if capacity is the problem

Reply:
will create issue with channel capacity config